spec/models/invoice_spec.rb

```diff
@@ -0,0 +1,14 @@
+require "rails_helper"
+
+RSpec.describe Invoice do 
+  describe "relationships" do 
+    it { should belong_to :customer }
+    it { should have_many :transactions }
+    it { should have_many :invoice_items }
+    it { should have_many(:items).through(:invoice_items) }
+  end
+
+  describe "validations" do 
+    it { should validate_presence_of :status }
```

Comment:
Same for here with the enums, something like

it { should define_enum_for(:status).with_values([completed: 0, cancelled: 1, in_progress: 2]) }

unsure on how the space is handled for "in progress"

Reply:
completed, no `[]`